src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1272,6 +1272,21 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 			return Position.ONE; // can only be ONE
 		}
 
+		const config = this.configurationService.getConfiguration<IWorkbenchEditorConfiguration>();
+		const reuseIfOpen = config.workbench.editor.reuseIfOpen;
+		// Respect option to reveal an editor if it is open (not necessarily already visible)
+		const skipReuse = (options && options.index) || arg1;
+		if(!skipReuse && reuseIfOpen) {
+			const groups = this.stacks.groups;
```

Comment:
@bpasero Sorry, my comment was related **only** to the value of extracting into a `findGroup` method, not adding tests for it. I did not mean to imply there's no value in adding tests.
Let's do it like this: I'll extract into a method and if you think it's readable, I'll proceed with the tests.